File: Neetcode 150/Binary Search/find-minimum-in-rotated-sorted-array.cpp

// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/

class Solution {
public:
    int findMin(vector<int>& nums) {
        
        int ans = INT_MAX;
        if(nums.size() == 0) return ans;

        int start = 0, end = nums.size()-1;
        while(start < end){
            int mid = start + (end-start)/2;
            if(nums[0] <= nums[mid]) start = mid+1;
            else end = mid;
        }

        return min(nums[0], nums[end]);
    }
};

File: Neetcode-150-js/Binary Search/find-minimum-in-rotated-sorted-array.js

// https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/

/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let len = nums.length;

    let start = 0, end = len-1;
    let minEle = Number.MAX_SAFE_INTEGER;
    while(start <= end){
        let mid = Math.floor((start+end)/2);
        
        if(nums[start] <= nums[mid]){
            minEle = Math.min(minEle, nums[start]);
            start = mid+1;
        }
        // nums[mid] <= nums[end]
        else{
            minEle = Math.min(minEle, nums[mid]);
            end = mid-1;
        }
    }

    return minEle;
};